pss/pss.go

```diff
@@ -60,3 +67,27 @@ func (p *Pss) Send(ctx context.Context, targets [][]byte, topic trojan.Topic, pa
 
 	return tc, nil
 }
+
+// Register allows the definition of a Handler func for a specific topic on the pss struct
+func (p *Pss) Register(topic trojan.Topic, hndlr Handler) {
+	p.handlersMu.Lock()
+	defer p.handlersMu.Unlock()
+	p.handlers[topic] = hndlr
+}
+
+// Deliver allows unwrapping a chunk as a trojan message and calling its handler func based on its topic
+func (p *Pss) Deliver(c chunk.Chunk) error {
```

Comment:
add a test for this

Reply:
added `TestRegister` and `TestDeliver` func (the former also implicitely tests `pss.getHandler`).

this led me to descover a bug in `trojan.NewMessage`, so i've extended its test as well.

thanks